### PR TITLE
[Fix] modify developer guide doc

### PR DESCRIPTION
**Pull request by RandyLambert**
* modify developer guide doc

Signed-off-by: RandyLambert <randylambert@foxmail.com>

### CODE REVIEW

File: docs/extend-chaos-daemon-interface.md [hunk 1]
```diff
@@ -75,36 +78,37 @@ To allow Chaos Daemon to accept the requests from Chaos Controller Manager, you
...
-   out, err := cmd.Output()
-   if err != nil {
-       return nil, err
-   }
-   if len(out) != 0 {
-       log.Info("cmd output", "output", string(out))
-   }
-
-   return &empty.Empty{}, nil
+      log := s.getLoggerFromContext(ctx)
```
Comment (fewdan) [inline docs/extend-chaos-daemon-interface.md:91]: It seems that there is a small problem with indentation here?

File: docs/extend-chaos-daemon-interface.md [hunk 2]
```diff
@@ -120,72 +124,74 @@ To allow Chaos Daemon to accept the requests from Chaos Controller Manager, you
...
-
-   _, err = pbClient.ExecHelloWorldChaos(ctx, &pb.ExecHelloWorldRequest{
-       ContainerId: containerId,
-   })
-   if err != nil {
-       return v1alpha1.NotInjected, err
-   }
-
-   return v1alpha1.Injected, nil
+      impl.Log.Info("Apply helloworld chaos")
```
Comment (fewdan) [inline docs/extend-chaos-daemon-interface.md:146]: ditto

**Comment by iguoyr** [comment]
Please replace `chaos-testing` with `chaos-mesh`:
https://github.com/chaos-mesh/website/blob/5b4916654d04f08d513178ef5a0470a11a2d5f6a/docs/add-new-chaos-experiment-type.md?plain=1#L274-L317

File: docs/add-new-chaos-experiment-type.md [hunk 3]
```diff
@@ -258,7 +241,13 @@ After you update the template, try running HelloWorldChaos.
...
-2. Deploy Chaos Mesh:
+2. Create a namespace for installing Chaos Mesh. It is recommended to install Chaos Mesh under the chaos-testing namespace. You can also specify any namespace to install Chaos Mesh:
+
+   ```bash
+   kubectl create ns chaos-testing
+   ```
+
+3. Deploy Chaos Mesh:
 
    <PickHelmVersion className="language-bash">{`helm install chaos-mesh helm/chaos-mesh --namespace=chaos-testing --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock --version latest`}</PickHelmVersion>
```
[the remark this answers is not in the document]
Comment (RandyLambert) [reply docs/add-new-chaos-experiment-type.md]: > 

done

**Comment by RandyLambert** [comment]
> Please replace `chaos-testing` with `chaos-mesh`:
> 
> https://github.com/chaos-mesh/website/blob/5b4916654d04f08d513178ef5a0470a11a2d5f6a/docs/add-new-chaos-experiment-type.md?plain=1#L274-L317

chaos-mesh/website has a lot of chaos-testing namespace. If we want to change it to chaos-mesh, I think we need to change all the chaos-testing in the website?

**Comment by iguoyr** [comment]
> > Please replace `chaos-testing` with `chaos-mesh`:
> > https://github.com/chaos-mesh/website/blob/5b4916654d04f08d513178ef5a0470a11a2d5f6a/docs/add-new-chaos-experiment-type.md?plain=1#L274-L317
> 
> chaos-mesh/website has a lot of chaos-testing namespace. If we want to change it to chaos-mesh, I think we need to change all the chaos-testing in the website?

@RandyLambert Yes, I create an issue #326, and I think we could fix it in #326, so you could ignore the comment I wrote before :)

File: docs/add-new-chaos-experiment-type.md [hunk 4]
```diff
@@ -222,23 +221,7 @@ You need to register the CRD (Custom Resource Definition) of HelloWorldChaos to
...
-chaosDaemon:
-  image: localhost:5000/chaos-mesh/chaos-daemon:latest
-  ...
-dashboard:
-  image: localhost:5000/chaos-mesh/chaos-dashboard:latest
-  ...
-```
-
-After you update the template, try running HelloWorldChaos.
+Try running HelloWorldChaos.
```
Comment (STRRL) [inline docs/add-new-chaos-experiment-type.md:224]: how about also removing this line: "Try running HelloWorldChaos."?

File: docs/extend-chaos-daemon-interface.md [hunk 5]
```diff
@@ -75,36 +78,37 @@ To allow Chaos Daemon to accept the requests from Chaos Controller Manager, you
    package chaosdaemon
 
    import (
-   "context"
-   "fmt"
+   	"context"
+   	"fmt"
```
Comment (STRRL) [inline docs/extend-chaos-daemon-interface.md:82]:
```suggestion
   "context"
   "fmt"
```

File: docs/extend-chaos-daemon-interface.md [hunk 6]
```diff
@@ -75,36 +78,37 @@ To allow Chaos Daemon to accept the requests from Chaos Controller Manager, you
...
 
-   "github.com/golang/protobuf/ptypes/empty"
+   	"github.com/golang/protobuf/ptypes/empty"
```
Comment (STRRL) [inline docs/extend-chaos-daemon-interface.md:84]:
```suggestion
   "github.com/golang/protobuf/ptypes/empty"
```

File: docs/extend-chaos-daemon-interface.md [hunk 7]
```diff
@@ -120,72 +124,74 @@ To allow Chaos Daemon to accept the requests from Chaos Controller Manager, you
...
-   "sigs.k8s.io/controller-runtime/pkg/client"
+   	"context"
+
+   	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+   	impltypes "github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/types"
+   	"github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/utils"
+   	"github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
+   	"github.com/go-logr/logr"
+   	"go.uber.org/fx"
+   	"sigs.k8s.io/controller-runtime/pkg/client"
```
Comment (STRRL) [inline docs/extend-chaos-daemon-interface.md:135]:
```suggestion
   "context"

   "github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
   impltypes "github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/types"
   "github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/utils"
   "github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
   "github.com/go-logr/logr"
   "go.uber.org/fx"
   "sigs.k8s.io/controller-runtime/pkg/client"
```

File: docs/extend-chaos-daemon-interface.md [hunk 8]
```diff
@@ -120,72 +124,74 @@ To allow Chaos Daemon to accept the requests from Chaos Controller Manager, you
...
+   	containerId := decodedContainer.ContainerId
+
+   	_, err = pbClient.ExecHelloWorldChaos(ctx, &pb.ExecHelloWorldRequest{
+   		ContainerId: containerId,
+   	})
+   	if err != nil {
+   		return v1alpha1.NotInjected, err
+   	}
+
+   	return v1alpha1.Injected, nil
```
Comment (STRRL) [inline docs/extend-chaos-daemon-interface.md:161]:
```suggestion
   impl.Log.Info("Apply helloworld chaos")
   decodedContainer, err := impl.decoder.DecodeContainerRecord(ctx, records[index], obj)
   if err != nil {
   	return v1alpha1.NotInjected, err
   }
   pbClient := decodedContainer.PbClient
   containerId := decodedContainer.ContainerId

   _, err = pbClient.ExecHelloWorldChaos(ctx, &pb.ExecHelloWorldRequest{
       ContainerId: containerId,
   })
   if err != nil {
       return v1alpha1.NotInjected, err
   }

   return v1alpha1.Injected, nil
```

File: docs/add-new-chaos-experiment-type.md [hunk 9]
```diff
@@ -111,36 +111,35 @@ You need to register the CRD (Custom Resource Definition) of HelloWorldChaos to
...
        impl.Log.Info("Hello world!")
        return v1alpha1.Injected, nil
    }
 
-   // This corresponds to the Recover phase of HelloWorldChaos. The reconciler will be triggered to recover the chaos action.
+   // Recover means the reconciler recovers the chaos action
    func (impl *Impl) Recover(ctx context.Context, index int, records []*v1alpha1.Record, obj v1alpha1.InnerObject) (v1alpha1.Phase, error) {
        impl.Log.Info("Goodbye world!")
        return v1alpha1.NotInjected, nil
    }
```
[the remark this answers is not in the document]
Comment (RandyLambert) [reply docs/add-new-chaos-experiment-type.md:134]: > What about using similar pattern in the comments for `Apply` and `Recover`?

done

File: docs/add-new-chaos-experiment-type.md [hunk 10]
```diff
@@ -111,36 +111,35 @@ You need to register the CRD (Custom Resource Definition) of HelloWorldChaos to
        "sigs.k8s.io/controller-runtime/pkg/client"
 
        "github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+        impltypes "github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/types"
```
[the remark this answers is not in the document]
Comment (RandyLambert) [reply docs/add-new-chaos-experiment-type.md]: > 

done

**Comment by Oreoxmt** [comment]
@iguoyr @STRRL @fewdan PTAL again

**Review by STRRL — APPROVED** [review]
LGTM

**Comment by Oreoxmt** [comment]
@iguoyr @fewdan PTAL again

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**